source/dagger/gameplay/platformer/platformer_collision.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+#include "core/system.h"
+#include "core/core.h"
+#include <array>
+
+#include "gameplay/platformer/platformer_controller.h"
+
+using namespace dagger;
+using namespace platformer;
+
+struct CollisionInfo
+{
+    Bool hasCollided{ false };
+    // 0: right | 1: left | 2: top  |  3: down
+    std::array<Bool, 4> collisionSide{ false, false,false,false };
```

Comment:
`StaticArray<Bool, 4>` is the Dagger way to go :) if we replace this with something in the future, it'll be easier to use the abstracted version.